aloha-core/src/main/scala/com/eharmony/aloha/models/multilabel/MultilabelModel.scala

```diff
@@ -0,0 +1,334 @@
+    Subvalue(aud, None)
+  }
+
+  /**
+    *
+    * @param modelId
+    * @param labelInfo
+    * @param auditor
+    * @tparam U
+    * @tparam K
+    * @tparam B
+    * @return
+    */
+  protected[multilabel] def reportNoPrediction[U, K, B <: U](
+      modelId: ModelIdentity,
+      labelInfo: LabelsAndInfo[K],
+      auditor: Auditor[U, Map[K, Double], B]
+  ): Subvalue[B, Nothing] = {
+    // TODO: Fill in the errors.
+    val aud = auditor.failure(modelId)
+    Subvalue(aud, None)
+  }
+
+  /**
+    *
+    * @param modelId
+    * @param labelInfo
+    * @param missing
+    * @param prediction
+    * @param auditor
+    * @tparam U
+    * @tparam K
+    * @tparam B
+    * @return
+    */
+  protected[multilabel] def reportSuccess[U, K, B <: U](
+      modelId: ModelIdentity,
+      labelInfo: LabelsAndInfo[K],
+      missing: scm.Map[String, Seq[String]],
+      prediction: Map[K, Double],
+      auditor: Auditor[U, Map[K, Double], B]
+  ): Subvalue[B, Map[K, Double]] = {
+
+    val errors =
+      if (labelInfo.missingLabels.nonEmpty)
+        Seq(s"Labels provide for which a prediction could not be produced: ${labelInfo.missingLabels.mkString(", ")}.")
+      else Seq.empty
+
+    // TODO: Incorporate missing data reporting.
+    val aud: B = auditor.success(modelId, prediction, errorMsgs = errors)
+
+    Subvalue(aud, Option(prediction))
+  }
+
+  /**
+    *
+    * @param modelId
+    * @param labelInfo
+    * @param missing
+    * @param throwable
+    * @param auditor
+    * @tparam U
+    * @tparam K
+    * @tparam B
+    * @return
+    */
+  protected[multilabel] def reportPredictorError[U, K, B <: U](
+      modelId: ModelIdentity,
+      labelInfo: LabelsAndInfo[K],
+      missing: scm.Map[String, Seq[String]],
+      throwable: Throwable,
+      auditor: Auditor[U, Map[K, Double], B]
+  ): Subvalue[B, Nothing] = {
+
+    // TODO: Fill in.
+    val aud = auditor.failure(modelId)
+    Subvalue(aud, None)
+  }
+
+  /**
+    *
+    * @param a
+    * @param labelsOfInterest
+    * @param labelToInd
+    * @tparam A
+    * @tparam K
+    * @return
+    */
+  protected[multilabel] def labelsForPrediction[A, K](
+      a: A,
+      labelsOfInterest: GenAggFunc[A, sci.IndexedSeq[K]],
+      labelToInd: Map[K, Int]
+  ): LabelsAndInfo[K] = {
+
+    val labelsShouldPredict = labelsOfInterest(a)
+
+    val unsorted =
+      for {
+        label <- labelsShouldPredict
+        ind   <- labelToInd.get(label).toList
+      } yield (ind, label)
+
+    val problems =
+      if (labelsShouldPredict.nonEmpty) None
+      else Option(labelsOfInterest.accessorOutputProblems(a))
+
+    val noPrediction =
+      if (unsorted.size == labelsShouldPredict.size) Seq.empty
+      else labelsShouldPredict.filterNot(labelToInd.contains)
+
+    val (ind, lab) = unsorted.sortBy{ case (i, _) => i }.unzip
+
+    LabelsAndInfo(ind, lab, noPrediction, problems)
+  }
+
+
+  override def parser: ModelParser = Parser
```

Comment:
Please disregard the `Parser` **object** as it is just a stub for this phase of development.